_posts/blog/2016-10-28-durability-performance.md

```diff
@@ -0,0 +1,136 @@
+for configuring durability.  It also stopped using the `tserver.wal.sync.method`
+property.  The `table.durability` property has the following four legal values.
+
+ * **none** : Do not write to WAL            
+ * **log**  : Write to WAL, but do not sync  
+ * **flush** : Write to WAL and call `hflush` 
+ * **sync** : Write to WAL and call `hsync`  
+
+If multiple writes arrive at around the same time with different durability
+settings, then the group commit code will choose the most conservative
+durability.  This can cause one tables settings to slow down writes to another
+table.  
+
+In Accumulo 1.6, it was easy to make all writes use `hflush` because there was
+only one tserver setting.  Getting everything to use `flush` in 1.7 and later
+can be a little tricky because by default the Accumulo metadata table is set to
+use `sync`.  Executing the following command in the Accumulo shell will
+accomplish this (assuming no tables or namespaces have been specifically set to
+`sync`).  The first command sets a system wide table default for `flush`.  The
+second two commands override metadata table specific settings of `sync`.
+
+```
+config -s table.durability=flush
+config -t accumulo.metadata -s table.durability=flush
+config -t accumulo.root -s table.durability=flush
+```
+
+Even with these settings adjusted, minor compactions could still force `hsync`
+to be called in 1.7.0 and 1.7.1.  This was fixed in 1.7.2 and 1.8.0.  See the
+[1.7.2 release notes][172_RN_MCHS] and [ACCUMULO-4112] for more details.
+
+In addition to the per table durability setting, a per batch writer durability
+setting was also added in 1.7.0.  See
+[BatchWriterConfig.setDurability(...)][SD].  This means any client could
+potentially cause a `hsync` operation to occur, even if the system is
+configured to use `hflush`.
+
+## Improving the situation
+
+The more granular durability settings introduced in 1.7.0 can cause some
+unexpected problems.  [ACCUMULO-4146] suggest one possible way to solve these
+problems with Per-durability write ahead logs.
+
+[fcf]: https://docs.oracle.com/javase/8/docs/api/java/nio/channels/FileChannel.html#force-boolean-
+[ros1]: https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/ReplicaOutputStreams.java#L78
+[ros2]: https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/ReplicaOutputStreams.java#L87
+[fos]: https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockReceiver.java#L358
+[ACCUMULO-4146]: https://issues.apache.org/jira/browse/ACCUMULO-4146
+[ACCUMULO-4112]: https://issues.apache.org/jira/browse/ACCUMULO-4112
+[160_RN_WAL]: /release_notes/1.6.0#slower-writes-than-previous-accumulo-versions
```

Comment:
Jekyll note: you should prefix absolute links with `{{ site.baseurl }}` so the links don't break if running locally or on a test fork.